cyral/model_integration_logging.go

```diff
@@ -108,20 +129,26 @@ func getIntegrationLogsSchema() map[string]*schema.Schema {
 						Elem: &schema.Resource{
 							Schema: map[string]*schema.Schema{
 								"region": {
-									Description: "The AWS region.",
+									Description: "AWS region.",
 									Required:    true,
 									Type:        schema.TypeString,
 								},
 								"group": {
-									Description: "The CloudWatch log group.",
+									Description: "CloudWatch log group.",
 									Required:    true,
 									Type:        schema.TypeString,
 								},
 								"stream": {
-									Description: "The CloudWatch log stream.",
-									Required:    true,
+									Description: "CloudWatch log stream.",
+									Optional:    true,
 									Type:        schema.TypeString,
 								},
+								"log_retention_days": {
+									Description:  "Log retention days (optional). If specified, valid values are [1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653]. If unspecified, CloudWatch will retain the logs indefinitely.",
+									Optional:     true,
+									Type:         schema.TypeInt,
+									ValidateFunc: validateLogRetentionDays,
```

Comment:
Would be better to delegate this validation to the API, since it already validates this field and if we ever need to change the validation in the future it would be easier to maintain it in a single place. I understand that adding the validation directly in the terraform scheme improves the resource UX a little bit, since the error will be caught earlier by the terraform instead of during the runtime of the API, so, in the end, is a tradeoff between a slight improvement in the UX vs maintenance, and I think in this case its better to opt for the latter. This means that the `validateLogRetentionDays` function can be removed.

Reply:
Addr [bb779d9](https://github.com/cyralinc/terraform-provider-cyral/pull/395/commits/bb779d978dffb1a275a7d719ea74685ee6f3e497)